src/NuGetForUnity/Editor/TargetFrameworkResolver.cs

```diff
@@ -91,20 +96,12 @@ internal static class TargetFrameworkResolver
         };
 
         private static DotnetVersionCompatibilityLevel CurrentBuildTargetDotnetVersionCompatibilityLevel
-        {
-            get
+            => CurrentApiCompatibilityLevel switch
```

Comment:
Aa far as I know switch expressions are not supported in Unity 2018 that we are still supporting. You can test this and other of your changes if you open src/NuGetForUnity.Tests project in Unity 2018.4.36f1.